hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -623,7 +630,7 @@ public CascadeStyle getCascadeStyle(int i) {
 
 	@Override
 	public boolean isMutable() {
-		return compositeUserType == null || compositeUserType.isMutable();
+		return (compositeUserType == null || compositeUserType.isMutable()) && !ReflectHelper.isRecord( componentClass );
```

Comment:
IMO this should be stored in a field in `ComponentType`, as the `isRecord` check is not cheap.

Reply:
You are right, I will fix that.